File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
'''
비상금 숨겨놓는 장소 => 비밀지도
비밀 지도 암호 해독하기
한 변의 길이가 n인 정사각형 배열 형태, 각 칸은 공백, 벽(#) 두 종류
전체 지도는 두 장의 지도를 겹쳐서 얻을 수 있다. 
어느 하나라도 벽인 곳은 전체 지도에서도 벽 |
두 지도 모두에서 공백인 부분은 전체 지도에서도 공백 & 
0, 1 => 1
1, 1 => 1
0, 0 => 0

1 => 벽
0 => 공백 

'''
    
def solution(n, arr1, arr2):
    answer = []
    arr1_list = []
    arr2_list = []
    for i in range(n):
        value = bin(arr1[i])[2:]
        if(len(value) < n):
            value = "0"*(n-len(value)) + value
        arr1_list.append(value) # string
        
    for i in range(n):
        value = bin(arr2[i])[2:] # 2진법으로 변환 
        if(len(value) < n):
            value = "0"*(n-len(value)) + value
        arr2_list.append(value) # string
    
    
    for i in range(n): # 0,1,2,3,4
        word = ""
        for j in range(len(arr1_list[i])):
            for k in range(len(arr2_list[i])):
                if(j == k):
                    if(arr1_list[i][j] == "0" and arr2_list[i][j] == "0"):
                        word += " "
                    elif(arr1_list[i][j] == "1" or arr2_list[i][j] == "1"):
                        word += "#"
        answer.append(word)
        
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        String[] arr1_bin = new String[n];
        String[] arr2_bin = new String[n];
        // 2진법으로 바꾸기, 길이가 n이 안되면 0으로 채우기 
        for(int i = 0; i < n; i++) {
        	String bin1 = Integer.toBinaryString(arr1[i]);
        	String new_arr1 = "";
        	if(bin1.length() < n) {
        		for(int j = 0; j < n-bin1.length();j++) {
        			new_arr1 += "0";
        		}
        	}
        	arr1_bin[i] = new_arr1 + bin1;
        	
        }
        
        for(int i = 0; i < n; i++) {
        	String bin2 = Integer.toBinaryString(arr2[i]);
        	String new_arr2 = "";
        	if(bin2.length() < n) {
        		for(int j = 0; j < n-bin2.length();j++) {
        			new_arr2 += "0";
        		}
        	}
        	arr2_bin[i] = new_arr2 + bin2;
        }
        // 여기서부터 문제,, ㅜㅠ
        // replace
        for(int i = 0; i < n; i++) {
        	String word = "";
        	for(int j = 0; j < arr1_bin[i].length(); j++) {
        		if(arr1_bin[i].charAt(j) == '0' && arr2_bin[i].charAt(j) == '0') {
        			word += " ";
        		}
        		else if(arr1_bin[i].charAt(j) == '1' || arr2_bin[i].charAt(j) == '1') {
        			word += "#";
        		}
        	}
        	answer[i] = word;
        }
        
        
        return answer;
    }
}